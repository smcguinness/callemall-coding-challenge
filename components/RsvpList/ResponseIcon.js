import React from 'react';
import PropTypes from 'prop-types';
import pluralize from 'pluralize';

import { Tooltip } from '@material-ui/core';

import DoneIcon from '@material-ui/icons/Done';
import DoneAllIcon from '@material-ui/icons/DoneAll';
import CloseIcon from '@material-ui/icons/Close';

const ResponseIcon = ({ guests, response }) => {
  if (response !== 'no') {
    const color = (response === 'yes' ? 'green' : 'grey');
    if (guests > 0) {
      return (
        <Tooltip title={pluralize('guest', guests, true)}>
          <DoneAllIcon style={{ color }} />
        </Tooltip>
      );
    }
    return <DoneIcon style={{ color }} />;
  }
  return <CloseIcon style={{ color: 'red' }} />;
};

ResponseIcon.defaultProps = {
  guests: 0,
  response: 'no',
};

ResponseIcon.propTypes = {
  guests: PropTypes.number,
  response: PropTypes.string,
};

export default ResponseIcon;
