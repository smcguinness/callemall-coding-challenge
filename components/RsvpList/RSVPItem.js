import React from 'react';
import PropTypes from 'prop-types';

import {
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
} from '@material-ui/core';

import ResponseIcon from './ResponseIcon';

const RSVPItem = ({ guests, member: { id, name }, response }) => (
  <ListItem alignItems="flex-start" key={id}>
    <ListItemText primary={name} />
    <ListItemSecondaryAction>
      <ResponseIcon guests={guests} response={response} />
    </ListItemSecondaryAction>
  </ListItem>
);

RSVPItem.defaultProps = {
  guests: 0,
};

RSVPItem.propTypes = {
  guests: PropTypes.number,
  member: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  }).isRequired,
  response: PropTypes.string.isRequired,
};

export default RSVPItem;
