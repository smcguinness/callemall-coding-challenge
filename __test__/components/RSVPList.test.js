import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import sinon from 'sinon';

import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';

import RSVPList from '../../components/RsvpList';
import ResponseIcon from '../../components/RsvpList/ResponseIcon';
import RSVPItem from '../../components/RsvpList/RSVPItem';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('RSVP List', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<RSVPList />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('RSVP Item', () => {
  let wrapper;
  beforeAll(() => {
    const props = {
      guests: 1,
      member: {
        id: 12345,
        name: 'John Smith',
      },
      response: 'yes',
    };
    wrapper = shallow(<RSVPItem {...props} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Response Icon', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<ResponseIcon />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
