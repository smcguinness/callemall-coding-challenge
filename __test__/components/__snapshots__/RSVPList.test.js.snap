// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSVP Item renders correctly 1`] = `
<WithStyles(ListItem)
  alignItems="flex-start"
  key="12345"
>
  <WithStyles(ListItemText)
    primary="John Smith"
  />
  <WithStyles(ListItemSecondaryAction)>
    <ResponseIcon
      guests={1}
      response="yes"
    />
  </WithStyles(ListItemSecondaryAction)>
</WithStyles(ListItem)>
`;

exports[`RSVP List renders correctly 1`] = `
<RsvpList
  classes={
    Object {
      "card": "RsvpList-card-1",
    }
  }
  rsvps={Array []}
/>
`;

exports[`Response Icon renders correctly 1`] = `
<pure(CloseIcon)
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
